pkg/infra/block_collector.go

```diff
@@ -18,54 +18,68 @@ type BlockCollector struct {
 	thresholdP, totalP int
 	totalTx            int
 	registry           map[uint64]int
+	addressBlock       map[uint64]map[string]bool
+}
+
+// BlockAddress describe the source of block
+type BlockAddress struct {
```

Comment:
`AddressedBlock` maybe?

and if we declare it with embedding:
```go
type AddressedBlock struct {
  *peer.FilteredBlock

  Address string
}
```
we could avoid some text changes in other part of code?